packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1041,10 +1112,154 @@ class HotRunner extends ResidentRunner {
     globals.printTrace('reloaded $loadedLibraryCount of $finalLibraryCount libraries');
     // reloadMessage = 'Reloaded $loadedLibraryCount of $finalLibraryCount libraries';
     // Record time it took for the VM to reload the sources.
-    _addBenchmarkData('hotReloadVMReloadMilliseconds', vmReloadTimer.elapsed.inMilliseconds);
+    hotRunner._addBenchmarkData('hotReloadVMReloadMilliseconds', vmReloadTimer.elapsed.inMilliseconds);
     return OperationResult(0, 'Reloaded $loadedLibraryCount of $finalLibraryCount libraries');
   }
 
+  Future<List<Future<vm_service.ReloadReport>>> _reloadDeviceSources(
+    FlutterDevice device,
+    String entryPath, {
+    bool pause = false,
+  }) async {
+    final String deviceEntryUri = device.devFS.baseUri
+      .resolve(entryPath).toString();
+    final vm_service.VM vm = await device.vmService.service.getVM();
+    return <Future<vm_service.ReloadReport>>[
+      for (final vm_service.IsolateRef isolateRef in vm.isolates)
+        device.vmService.service.reloadSources(
+          isolateRef.id,
+          pause: pause,
+          rootLibUri: deviceEntryUri,
+        )
+    ];
+  }
+
+  void _resetDevFSCompileTime(List<FlutterDevice> flutterDevices) {
+    for (final FlutterDevice device in flutterDevices) {
+      device.devFS.resetLastCompiled();
+    }
+  }
+}
+
+@visibleForTesting
+class ReassembleResult {
+  ReassembleResult(this.reassembleViews, this.failedReassemble, this.shouldReportReloadTime);
+  final Map<FlutterView, FlutterVmService> reassembleViews;
+  final bool failedReassemble;
+  final bool shouldReportReloadTime;
+}
+
+@visibleForTesting
+class ReassembleHelper {
```

Comment:
Since this class doesn't have any state or dependencies, you could instead use a function: `typedef ReassembleHelper = Future<ReassembleResult> reassemble(...)` and then inject that

Reply:
Done.